tests/cases/fourslash/stringLiteralCompletionsWithinInferredObjectWhenItsKeysAreUsedOutsideOfIt.ts

```diff
@@ -0,0 +1,33 @@
+/// <reference path="fourslash.ts" />
+
+// @strict: true
+
+//// declare function createMachine<T>(config: {
+////   initial: keyof T;
+////   states: {
+////     [K in keyof T]: {
+////       on?: Record<string, keyof T>;
+////     };
+////   };
+//// }): void;
+////
+//// createMachine({
+////   initial: "a",
+////   states: {
+////     a: {
+////       on: {
+////         NEXT: "/*1*/",
+////       },
+////     },
+////     b: {
+////       on: {
+////         NEXT: "/*2*/",
+////       },
+////     },
+////   },
+//// });
+
+verify.completions({
+    marker: ["1", "2"],
```

Comment:
It didn't work for marker 1 because the completions type took priority over the regular contextual type. With inference blocked and all the `T` was inferred here as `{ a: any }` based on the `initial: keyof T`. That result "locked" the string literal completions to that blocked result whereas the intended/better completions can be returned here based on the contextual type (when the inferred type of `T` is `{ a: unknown; b: unknown; }`).

Note that this PR doesn't improve the very same situation for regular (non-string) completions - when we have a node with `ThisNodeHasError` flag. Improvements to that can be explored separately.